### PR TITLE
add build and lint workflow in CI, fix lint errors

### PR DESCRIPTION
**Pull request by EvanBoyle**
- configure es lint
- fix errors
- Add CI with lint + build step

### CODE REVIEW

**Review by jmoseley — APPROVED** [review]
🔥

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    runs-on: ubuntu-latest
+    if: github.repository == 'cortexclick/cortex-sdk'
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20'
```
Comment (jmoseley) [inline .github/workflows/ci.yml:22]:
```suggestion
        with:
          node-version: '20'
          cache: 'npm'
```

File: package-lock.json
Comment (chaosrealm) [inline package-lock.json:1]: Should we .gitignore all package locks?

Comment (EvanBoyle) [reply package-lock.json:1]: No, you are supposed to commit it: https://docs.npmjs.com/cli/v8/configuring-npm/package-lock-json

Comment (jmoseley) [reply package-lock.json:1]: It is best practice to commit package lock to the repo. See https://docs.npmjs.com/cli/v10/configuring-npm/package-lock-json#see-also:~:text=This%20file%20is%20intended%20to%20be%20committed%20into%20source%20repositories

Comment (chaosrealm) [reply package-lock.json:1]: Got it, tks!

**Pull request merged**